shell/common/gin_converters/callback_converter.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2019 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef SHELL_COMMON_GIN_CONVERTERS_CALLBACK_CONVERTER_H_
+#define SHELL_COMMON_GIN_CONVERTERS_CALLBACK_CONVERTER_H_
+
+#include "gin/converter.h"
+#include "gin/function_template.h"
+
+// TODO(deermichel): check if this is a good idea?!
```

Comment:
I think we should either add a utility function with name like `ConvertCallbackToV8Leaked`, or use the non-leak version in native-mate.

Reply:
A function named `*Leaked` doesn't make me feel very good 😄. So for clarification, `CreateFunctionTemplate` caches each template and therefore is leaky - that makes it unsuitable for a converter where things might happen without the user noticing.

However, using it here (as part of a utility function) shouldn't hurt anybody since `Initialize` is only called once per Electron instance, right? 

For objects, this is indeed problematic because the leaky method would get called for every object instantiation. I have an idea for a solution which i will push in another PR if that's ok 😬.

Reply:
It is fine in this case, but not when people start using it for arbitrary callbacks. We must explicitly warn them that this functions leaks.

Having `Leak` in names is fine, Chromium has some commonly used utilities with this style, for example `LeakyLazyInstanceTraits` and `LeakySingletonTraits`, and I think it helps write correct code.